### PR TITLE
IsNormal matrix

### PR DESCRIPTION
**Pull request by Beakerboy**
If a matrix is normal, it is diagonalizable, and has eigenvalues and eigenvectors

### CODE REVIEW

**Comment by markrogoyski** [comment]
Thanks for the pull request. Nice addition.

Just curious. Is it expected that this will just work with an ObjectMatrix of Complex numbers? I think it would need to check the conjugate transpose of that. I didn't see any complex number unit tests, and I presume complex matrices are just not a public-facing feature yet, which is fine. I want to make sure I have the same understanding you do so I don't write some unit tests for something that isn't expected to work yet.

Thanks!

**Comment by Beakerboy** [comment]
Eventually, sure. Since we don’t have a ComplexNumber Class yet I didn’t write any tests for that. I’ve been treating the existing Matrix class as a NumericalMatrix. When the Matrix refactor goes through, i’ll spend some time either abstracting functions to work in different contexts, or writing different versions for each class.

**Comment by markrogoyski** [comment]
Cool. I just wanted to make sure we had the same understanding. I like that thought, that the Matrix is the NumericMatrix. Makes it easy. Thanks.

File: src/LinearAlgebra/Matrix.php [hunk 1]
```diff
@@ -1035,6 +1036,31 @@ public function isOrthogonal(): bool
...
+     * @throws Exception\MathException
+     */
+    public function isNormal(): bool
+    {
+        if (!$this->isSquare()) {
+            return false;
+        }
+
+        // AAᵀ = AᵀA
+        $I   = MatrixFactory::identity($this->m);
```
Comment (markrogoyski) [inline src/LinearAlgebra/Matrix.php]: Why do you need the identity matrix here? Doesn't look like it is used.

Comment (Beakerboy) [reply src/LinearAlgebra/Matrix.php]: Sorry, I just copied the function above and (poorly) modified it. An orthogonal Matrix is a special type of normal matrix.

**Pull request merged**